run_foundry.py

```diff
@@ -0,0 +1,478 @@
+from pyk.prelude.kint import leInt
+from pyk.prelude.ml import mlEqualsTrue
+
+from run_elrond_tests import *
+
+INPUT_FILE_NAME = 'foundry.json'
+
+def load_input_json(test_dir):
+    try:
+        with open(join(test_dir, INPUT_FILE_NAME), 'r') as f:
+            return json.load(f)
+    except FileNotFoundError as e:
+        raise FileNotFoundError('"{INPUT_FILE_NAME}" not found in "{test_dir}"')
+
+
+def load_wasm(filename):
+    with open(filename, 'rb') as f:
+        return wasm2kast.wasm2kast(f, filename)
+
+
+def find_test_wasm_path(test_dir):
+    test_wasm_path = glob.glob(test_dir + '/output/*.wasm')
+    # TODO this loads the first wasm file in the directory. what if there are multiple wasm files?
+    if test_wasm_path:
+        return test_wasm_path[0]
+    else:
+        raise ValueError(f'WASM file not found: {test_dir}/output/?.wasm')
+
+
+def load_contract_wasms(contract_wasm_paths):
+    contract_wasm_modules = {
+        bytes(f, 'ascii'): load_wasm(f) for f in contract_wasm_paths
+    }
+    return contract_wasm_modules
+
+
+def deploy_test(krun, test_wasm, contract_wasms):
+    """
+    1. create a main account: 'k'
+    2. reserve a new address for the test contract: owner = 'k', contract address = 'k-test'
+    3. deploy the test contract from account 'k': 'k-test'
+    """
+    # create the root account
+    k_addr = mandos_argument_to_kbytes('address:k')
+    init_main_acct = KApply(
+        'setAccount',
+        [
+            k_addr,
+            token(1),
+            token(100000),
+            KApply('.Code', []),
+            token(b''),
+            KMapBytesToBytes([]),
+        ],
+    )
+
+    # the test contract's address will be 'k-test'
+    k_test_addr = mandos_argument_to_kbytes('sc:k-test')
+    new_address = KApply('newAddress', [k_addr, token(1), k_test_addr])
+
+    # deploy the test contract
+    arguments = ListBytes(wrapBytes(token(k)) for k in contract_wasms)
+    gas = token(5000000000000)
+    deploy_cmd = KApply(
+        'deployTx', [k_addr, token(0), test_wasm, arguments, gas, token(0)]
+    )
+
+    # initialization steps
+    init_steps = KSequence([init_main_acct, new_address, deploy_cmd])
+
+    # create an empty config and embed init steps
+    empty_conf = krun.definition.init_config(KSort('GeneratedTopCell'))
+
+    conf, subst = split_config_from(empty_conf)
+    subst['K_CELL'] = init_steps
+    subst['WASMSTORE_CELL'] = map_of(
+        {token(path): mod for path, mod in contract_wasms.items()}
+    )
+    conf_with_steps = Subst(subst)(conf)
+
+    _, sym_conf, subst = run_config_and_check_empty(krun, conf_with_steps)
+
+    subst['WASMSTORE_CELL'] = map_of({})
+
+    return sym_conf, subst
+
+def run_config(krun: KRun, conf: KInner, log=False) -> CompletedProcess:
+    with krun._temp_file() as fkast:
+        conf_dict = { 'format': 'KAST', 'version': 2, 'term': conf.to_dict() }
+        json.dump(conf_dict, fkast)
+        fkast.flush()
+
+        kast_res = _kast(
```

Comment:
Why not `krun.kast_to_kore`, `krun.run_kore_term` and `krun.kore_to_kast` (or, at least, some of these)?

Reply:
These methods first do the conversion in Python, and if it fails [fall back](https://github.com/runtimeverification/pyk/blob/fd06d71322d85f41ecb652a3ac33d438c1bf0e4d/src/pyk/ktool/kprint.py#L271) to `kast`. I tried these methods first, but If the contract is big, this fails with too much recursion.

Reply:
Actually, this is only causing issues when running tests with concrete execution. It was only necessary for the demo. Claim generation works fine when implemented with these methods and a lower recursion limit. Do you think we should keep the code related to running tests with hypothesis?

Reply:
That's a product question, and I'm not good with those. We could ask Elrond about it...

Anyway, we could also increase the recursion limit here, not only in tests - what size would be needed?

I'm asking because, in my experience, kast_to_kore's errors are easier to understand, and with the Elrond semantics I didn't find yet a case in which the Python conversion fals, but kast succeeds.

Reply:
`run_elrond_tests.py` sets the recursion limit to a very high number but it turns out that hypothesis changes it. I've fixed it and refactored all uses of kast and krun with pyk methods. :+1: 

I'll keep the concrete execution with `hypothesis` part in case anyone wants to try it.